src/daemon/rpmostreed-deployment-utils.c

```diff
@@ -1112,7 +1025,12 @@ rpmostreed_update_generate_variant (OstreeSysroot *sysroot,
 
   /* but if there are no updates, then just ditch the whole thing and return NULL */
   if (is_new_checksum || rpmmd_modified_new)
-    *out_update = g_variant_ref_sink (g_variant_dict_end (&dict));
+    {
+      /* include a "state" checksum for cache invalidation; for now this is just the booted
+       * checksum (*not* base), though we could base it off more things later if needed */
+      g_variant_dict_insert (&dict, "state-sha512", "s", current_checksum);
```

Comment:
It's confusing to have the key named the same as the sack-based one.  Further it's not actually `sha512`...

Reply:
Also it's not the booted deployment, it's the merge deployment, right?

Reply:
Heh, that's where I took inspiration from. Clearly I was a bit too inspired :)
How about "update-sha256" ?